provider/marathon/marathon.go

```diff
@@ -233,14 +242,21 @@ func (p *Provider) taskFilter(task marathon.Task, application marathon.Applicati
 		return false
 	}
 
-	if _, err := processPorts(application, task); err != nil {
-		log.Errorf("Filtering Marathon task %s from application %s without port: %s", task.ID, application.ID, err)
-		return false
+	for _, serviceName := range p.getServiceNames(application) {
+		foundPort := false
+		if port := p.getPort(task, application, serviceName); len(port) > 0 {
+			foundPort = true
+		}
+
+		if !foundPort {
+			log.Errorf("Filtering Marathon task %s from application %s without port", task.ID, application.ID)
```

Comment:
One issue I see here is that if the application is missing an explicit port label for a service, it will fail.  The scenario where you would not have an app specific port, but say an app specific frontend.rule is valid otherwise.  So looping through all the available services to see if any of them has a port config was suppose to ensure that we find at least one port somewhere, which is enough to not filter the task.